### PR TITLE
Ukbb mri neck to knee ozery

### PR DESCRIPTION
**Pull request by michalozeryflato**

### CODE REVIEW

File: fuseimg/datasets/ukbb_neck_to_knee.py [hunk 1]
```diff
@@ -210,11 +213,15 @@ def dataset(
...
-        if is_female == None:
-            all_sample_ids = [file for file in os.listdir(data_dir) if '.zip' in file]
-        else:
-            all_sample_ids = [file for file in os.listdir(data_dir) if '.zip' in file]
-            all_sample_ids = list(set(input_source_gt[input_source_gt['is female'] == is_female][input_source_gt['file'].isin(all_sample_ids)]['file'].to_list()))
+
+        existing_files = [file for file in os.listdir(data_dir) if '.zip' in file]
+        existing_sample_id_fields = [f.split("_") for f in existing_files]
+        existing_sample_ids = set([a[0] + "_*_" + a[2] + "_" + a[3] for a in existing_sample_id_fields])
+        a_filter = input_source_gt['file'].isin(existing_sample_ids)
```
Comment (itaijj) [inline fuseimg/datasets/ukbb_neck_to_knee.py:220]: not sure why split and then merge again from existing_files  to existing_sample_ids .
you wanted to use later existing_sample_id_fields  for other fields? 
input_source_gt['file'] should be in which format?

File: fuseimg/datasets/ukbb_neck_to_knee.py [hunk 2]
```diff
@@ -210,11 +213,15 @@ def dataset(
...
+        if is_female is not None:
+            a_filter &= input_source_gt['is female'] == is_female
```
Comment (itaijj) [inline fuseimg/datasets/ukbb_neck_to_knee.py:222]: great idea on how to merge filter conditions .
if we have more complex situations ( like doing or or negation ) should we support?

File: fuseimg/datasets/ukbb_neck_to_knee.py (hunk 1 above)
Comment (michalozeryflato) [reply fuseimg/datasets/ukbb_neck_to_knee.py:220]: Not sure I understand the question.
existing_files =  a list filenames that exist on the disk. This names are NOT sample_ids
existing_sample_ids = sample_ids matching the existing files. These are of the form <eid>_*_<visit>_0
input_source_gt['file'] contains sample_id == file pattern = <eid>_*_<visit>_0

File: fuseimg/datasets/ukbb_neck_to_knee.py (hunk 2 above)
Comment (michalozeryflato) [reply fuseimg/datasets/ukbb_neck_to_knee.py:222]: I am changing this code  - to make it more generic.

**Review by itaijj — APPROVED** [review]

**Pull request merged**